src/Microsoft.DotNet.Interactive.Http/Model/HttpNamedRequest.cs

```diff
@@ -263,7 +269,32 @@ internal HttpNamedRequest(HttpRequestNode httpRequestNode, HttpResponse response
         JsonNode? newResponseJSON = null;
         try
         {
-            newResponseJSON = responseJSON[path[currentIndex]];
+            if (responseJSON is JsonObject j)
+            {
+
+                var pathMatch = arrayRegex.Match(path[currentIndex]);
+                if (pathMatch.Success)
+                {
+
+                    // Extract the array name and index
+                    string arrayName = pathMatch.Groups[1].Value;
+                    string indexString = pathMatch.Groups[2].Value;
+
+                    // Cast the index from string to int
+                    int index = int.Parse(indexString);
```

Comment:
`TryParse` here? A very long sequence of digits might not fit within a 32-bit int.